modules/openapi-generator/src/main/resources/ruby-client/gem.mustache

```diff
@@ -11,23 +11,14 @@ require '{{gemName}}/configuration'
 # Models
 {{#models}}
 {{#model}}
-{{^parent}}
-require '{{gemName}}/{{modelPackage}}/{{classFilename}}'
-{{/parent}}
-{{/model}}
-{{/models}}
-{{#models}}
-{{#model}}
-{{#parent}}
-require '{{gemName}}/{{modelPackage}}/{{classFilename}}'
-{{/parent}}
+{{moduleName}}.autoload :{{classname}}, '{{gemName}}/{{modelPackage}}/{{classFilename}}'
```

Comment:
There's a reason why we do it with the parent tags (to fix a bug reported previously about parent models not yet required) so please keep it that way.

Reply:
and likely we need an option for this (I can add it after merging this PR)

Reply:
Hi @wing328 ... thanks for the review.

> There's a reason why we do it with the parent tags (to fix a bug reported previously about parent models not yet required) so please keep it that way.

The parent tags were intentionally removed because they are no longer necessary.

The parent tags were introduced in https://github.com/OpenAPITools/openapi-generator/pull/9103 in order to fix https://github.com/OpenAPITools/openapi-generator/issues/4690, prior to which they didn't exist.  When using `require`, hierarchical ordering of the requires is required because the parent constant must exist, however when using `autoload`, Ruby will naturally load things in the correct hierarchical order regardless of the order of the autoload statements.  The fix in https://github.com/OpenAPITools/openapi-generator/pull/9103 was only a partial fix, because it only handled one level of inheritance, however, with autoload, we can support any level of inheritance.  Thus, this PR is a more robust alternative to https://github.com/OpenAPITools/openapi-generator/pull/9103, with the side benefit of improved memory and load time.

Here's an example where autoload doesn't care about the order, and also shows more than one level of inheritance:

Given:

```ruby
# foo.rb:
class Foo; end

# bar.rb:
class Bar < Foo; end

# baz.rb:
class Baz < Bar; end
```

Before:

```ruby
# main.rb
require "bar"
require "baz"
require "foo"
puts Baz.new
# => uninitialized constant Foo (NameError)
```

After:

```ruby
# main.rb
autoload :Bar, "bar"
autoload :Baz, "baz"
autoload :Foo, "foo"
puts Baz.new
# => #<Baz:0x00007fcebe0f1470>
```

> and likely we need an option for this (I can add it after merging this PR)

I'm not sure I understand the need for an option.  I look at this more like a more robust version of https://github.com/OpenAPITools/openapi-generator/pull/9103, which also didn't need an option as it was a bug fix.